test/catch2/live_restore/api/test_live_restore_fh_read_write.cpp

```diff
@@ -0,0 +1,216 @@
+{
+    auto file_name = "test_table.wt";
+    int allocsize = 4, page_size = allocsize * 4, file_size = 132;
+    auto dummy_char = '0';
+    auto src_char = '1';
+    auto write_char = '2';
+
+    live_restore_test_env env;
+    WT_SESSION *session = reinterpret_cast<WT_SESSION *>(env.session);
+    WT_SESSION_IMPL *session_impl = (WT_SESSION_IMPL *)session;
+    WTI_LIVE_RESTORE_FS *lr_fs = env.lr_fs;
+    WTI_LIVE_RESTORE_FILE_HANDLE *lr_fh = nullptr;
+
+    SECTION("Read/write when source exists")
+    {
+        create_file(env.source_file_path(file_name), file_size, src_char);
+        // Create a dest file, pre-filled dummy chars in the dest file is used to verify we are not
+        // touching them when their corresponding bits are set in the bitmap.
+        create_file(env.dest_file_path(file_name), file_size, dummy_char);
+
+        init_file_handle(
+          session, lr_fs, env.dest_file_path(file_name).c_str(), allocsize, file_size, &lr_fh);
+        char *write_buf = nullptr, *read_buf = nullptr;
+
+        // No writes yet, reads go into src.
+        REQUIRE(__wt_calloc(session_impl, 1, file_size + 1, &read_buf) == 0);
+        auto data = std::string(page_size, src_char);
+        REQUIRE(
+          lr_fh->iface.fh_read((WT_FILE_HANDLE *)lr_fh, session, 0, page_size, read_buf) == 0);
+        REQUIRE(WT_STRING_MATCH(read_buf, data.c_str(), page_size));
+
+        /*
+         * Background migration in progress, simulate this by calling a fh_write() to write src
+         * chars to dest, where the first few pages are fully migrated and the last page is
+         * partially migrated.
+         */
+        auto count = 13;
+        auto background_write_len = allocsize * count;
+        REQUIRE(background_write_len <= file_size);
+        data = std::string(background_write_len, src_char);
+        REQUIRE(__wt_calloc(session_impl, 1, background_write_len + 1, &write_buf) == 0);
+        REQUIRE(__wt_snprintf(write_buf, background_write_len + 1, "%s", data.c_str()) == 0);
+        REQUIRE(lr_fh->iface.fh_write(
+                  (WT_FILE_HANDLE *)lr_fh, session, 0, background_write_len, write_buf) == 0);
+
+        // The first floor(migration_page_count) are fully migrated and the data should have been
+        // copied to the dest file.
+        data = std::string(page_size, src_char);
+        int offset = 0;
+        for (; offset + page_size < background_write_len; offset += page_size) {
+            REQUIRE(lr_fh->iface.fh_read(
+                      (WT_FILE_HANDLE *)lr_fh, session, offset, page_size, read_buf) == 0);
+            REQUIRE(WT_STRING_MATCH(read_buf, data.c_str(), page_size));
+            // Verify data has been written to the dest file.
+            REQUIRE(lr_fh->destination->fh_read(
+                      lr_fh->destination, session, offset, page_size, read_buf) == 0);
+            REQUIRE(WT_STRING_MATCH(read_buf, data.c_str(), page_size));
```

Comment:
Can we use C++ string comparison here?

```suggestion
            REQUIRE(data == read_buf);
```

Reply:
Thanks, I've changed those buffers to use vector instead now.